src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final float GROW_FACTOR = 2f;
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table.length * LOAD_FACTOR < size) {
+            resize();
+        }
+        int hashKeyNode = hashKey(key);
+        int index = hashKeyNode & (table.length - 1);
+        Node<K, V> currentNode = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+        } else {
+            while ((currentNode.key != key && !currentNode.key.equals(key))
+                    && currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            if (currentNode.key == key || currentNode.key.equals(key)) {
```

Comment:
you use this code several times. extract it to separate function

Reply:
if (currentNode.key == key || currentNode.key.equals(key)) {
                return currentNode.value;
            }

if (currentNode.key == key || currentNode.key.equals(key)) {
                currentNode.value = value;
                return;
            }

The code has differences. In the second case, the keyword is return. Tell me how to write a method

Reply:
I believe Andriy wanted to make function for condition
```suggestion
            if (areEqual(key, currentNode.key)) {
```

Reply:
This is called overcoding, more operations are performed